### PR TITLE
Migrate select-variable directive to Vue

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

**This PR is based on https://github.com/matomo-org/tag-manager/pull/404.**

Changes:
* migrate piwik-variable-select directive to Vue.
* start adding type definitions for TagManager entity types.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: vue/src/VariableSelect/VariableSelect.vue [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+      <li class="collection-header">
+        <h4>{{ translate('TagManager_CustomVariables') }}</h4>
+      </li>
+      <li
+        class="collection-item"
+        @click="selectVariable(variable)"
+        v-for="(variable, index) in containerVariables"
+        :key="index"
+      >
+        <span class="title">{{ variable.name }} (Type: {{ variable.type }})</span>
```
Comment (sgiehl) [inline vue/src/VariableSelect/VariableSelect.vue]: We could maybe directly use the translation here, even though it was missing before ;-)

```suggestion
        <span class="title">{{ variable.name }} ({{ translate('TagManager_Type') }}: {{ variable.type }})</span>
```

File: vue/src/VariableSelect/VariableSelect.vue [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        <span class="secondary-content">
+          <i
+            class="icon-edit"
+            @click.prevent="editVariable(variable)"
```
Comment (sgiehl) [inline vue/src/VariableSelect/VariableSelect.vue]: Seems the `prevent` doesn't work as expected here. When clicking the edit icon, the edit dialog opens, but the variable is also selected in the background, which wasn't the case before.

**Review by sgiehl — CHANGES_REQUESTED** [review]
Left some comments. Besides that it seems to work as expected.

**Comment by diosmosis** [comment]
@sgiehl applied the fixes

**Review by sgiehl — APPROVED** [review]

**Pull request merged**